src/Database.cpp

```diff
@@ -0,0 +1,460 @@
+#include "Player.hpp"
+#include <list>
+#include "CNStructs.hpp"
+
+using namespace json11;
+
+//TODO: replace this sqlite wrapper with something better, clean up queries, get rid of json
+
+sqlite3pp::database db;
+
+void Database::open() {
+	//checking if database file exists
+	std::ifstream file;
+	file.open("data.db");
+	if (file) {
+		file.close();
+		//if exists, assign it		
+		db = sqlite3pp::database ("data.db");
+
+		DEBUGLOG(
+			std::cout << "Database in operation" << std::endl;
+		)
+	}
+	else {
+		//if doesn't, create all the tables
+		DEBUGLOG(
+			std::cout << "Creating new database" << std::endl;
+		)
+		db = sqlite3pp::database("data.db");
+		db.execute("CREATE TABLE Accounts(AccountID INTEGER PRIMARY KEY AUTOINCREMENT, Login TEXT NOT NULL, Password TEXT NOT NULL);");
+		db.execute("CREATE TABLE Players(PlayerID INTEGER PRIMARY KEY AUTOINCREMENT, AccountID INTEGER NOT NULL, Slot INTEGER NOT NULL, FirstName TEXT NOT NULL, LastName TEXT NOT NULL, CharData TEXT NOT NULL);");
+		DEBUGLOG(
+			std::cout << "Done" << std::endl;
+		)
+	}
+	
+
+}
+
+bool Database::isLoginDataGood(std::string login, std::string password) {
+	std::regex loginRegex("^([A-Za-z\\d_\\-]){5,20}$");
+	std::regex passwordRegex("^([A-Za-z\\d_\\-@$!%*#?&,.+:;<=>]){8,20}$");
+	return (std::regex_match(login, loginRegex)&&std::regex_match(password, passwordRegex ));
+}
+
+void Database::addAccount(std::string login, std::string password) {
+	sqlite3pp::command cmd(
+		db, "INSERT INTO Accounts (Login, Password) VALUES (:login, :password)"
+	);
+	password = BCrypt::generateHash(password);
+
+	cmd.bind(":login", login, sqlite3pp::nocopy);
+	cmd.bind(":password", password, sqlite3pp::nocopy);
+	cmd.execute();
+	
+}
+
+bool Database::doesUserExist(std::string login) {
+
+	std::string q = "SELECT COUNT(AccountID) FROM Accounts WHERE Login = \"" + login + "\"";
+	const char* query = q.c_str();
+	sqlite3pp::query qry(
+		db, query
+	);
+	
+	sqlite3pp::query::iterator i = qry.begin();
+	int result;
+	std::tie(result) =
+		(*i).get_columns<int>(0);
+	return (result != 0);
+}
+
+bool Database::isPasswordCorrect(std::string login, std::string password) {
+	std::string q = "SELECT Password FROM Accounts WHERE Login = \"" + login + "\"";
+	const char* query = q.c_str();
+	sqlite3pp::query qry(
+		db, query
+	);
+
+	sqlite3pp::query::iterator i = qry.begin();
+	const char* actual;
+	std::tie(actual) =
+		(*i).get_columns<const char*>(0);
+	return BCrypt::validatePassword(password, actual);
+		
+}
+
+int Database::getUserID(std::string login) {
+	std::string q = "SELECT AccountID FROM Accounts WHERE Login = \"" + login + "\"";
+	const char* query = q.c_str();
+	sqlite3pp::query qry(
+		db, query
+	);
+
+	sqlite3pp::query::iterator i = qry.begin();
+	int result;
+	std::tie(result) =
+		(*i).get_columns<int>(0);
+	return result;
+}
+
+int Database::getUserSlotsNum(int AccountId) {
+	std::string q = "SELECT COUNT(PlayerID) FROM Players WHERE AccountID = "+ std::to_string(AccountId);
```

Comment:
This is "theoretically fine" but should very much be using prepared statements as well.